pkg/controllers/state/cluster.go

```diff
@@ -173,9 +173,13 @@ func (c *Cluster) Nodes() StateNodes {
 	c.mu.RLock()
 	defer c.mu.RUnlock()
 
-	return lo.Map(lo.Values(c.nodes), func(n *StateNode, _ int) *StateNode {
-		return n.DeepCopy()
-	})
+	return StateNodes{
+		Nodes: lo.Map(lo.Values(c.nodes), func(n *StateNode, _ int) *StateNode {
```

Comment:
Do we need to create a new structure like this that causes us have to call into `Nodes` to get our nodes back? This also doesn't really indicate there is a client attached to it. This is back to my thoughts on `StateNode` being a data holder and not a client in and of itself. StateNodes is just a set of data holders.